src/Compilers/CSharp/Test/CommandLine/SarifV2ErrorLoggerTests.cs

```diff
@@ -289,12 +292,78 @@ internal override string GetExpectedOutputForAnalyzerDiagnosticsWithAndWithoutLo
                 AnalyzerForErrorLogTest.GetExpectedV2ErrorLogRulesText());
         }
 
+        internal string GetExpectedOutputForAnalyzerDiagnosticsWithJustfication(MockCSharpCompiler cmd, string justification)
+        {
+            string expectedOutput =
+@"{{
+  ""$schema"": ""http://json.schemastore.org/sarif-2.1.0"",
+  ""version"": ""2.1.0"",
+  ""runs"": [
+    {{
+{5},
+      ""tool"": {{
+        ""driver"": {{
+          ""name"": ""{0}"",
+          ""version"": ""{1}"",
+          ""dottedQuadFileVersion"": ""{2}"",
+          ""semanticVersion"": ""{3}"",
+          ""language"": ""{4}"",
+{6}
+        }}
+      }},
+      ""columnKind"": ""utf16CodeUnits""
+    }}
+  ]
+}}";
+            return FormatOutputText(
+                expectedOutput,
+                cmd,
+                AnalyzerForErrorLogTest.GetExpectedV2ErrorLogWithJustificationResultsText(cmd.Compilation, justification),
+                AnalyzerForErrorLogTest.GetExpectedV2ErrorLogRulesText());
+        }
+
         [ConditionalFact(typeof(WindowsOnly), Reason = "https://github.com/dotnet/roslyn/issues/30289")]
         public void AnalyzerDiagnosticsWithAndWithoutLocation()
         {
             AnalyzerDiagnosticsWithAndWithoutLocationImpl();
         }
 
+        [ConditionalFact(typeof(WindowsOnly))]
+        public void SimpleCompilerDiagnosticsSuppressedWithJustification()
```

Comment:
Need tests where:

1. Justification entry is missing 
1. Justification entry is empty